EIPS/eip-parallel_tx_map.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Parallel Execution Maps
+description: Optionally include list of transactions without dependencies.
+author: Justin Florentine (@jflo) <justin@florentine.us>
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-05-22
+requires: [EIP-7685](eip-7685.md)
+---
+
+## Abstract
+
+With the advent of parallel transaction execution in some execution clients, it is now possible to share the safe execution path with other clients. When a client that supports parallel execution composes a block, and they know the set of independent transactions which can be executed out of order, they can share that set with all other clients which can then benefit from a parallel execution path.
+
+## Motivation
+
+On average, up to 70% of the transactions in a block are often independent of all the others. This large subset of the block may be executed in any order and still arrive at the same state root. The difficult part for clients is determining which transactions these are. 
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+This EIP introduces a new EIP-7685 request type `0x03` that clients MUST NOT treat as an error upon encountering.
+
+| Name                     | Value  | Comment                                                       |
+|--------------------------|--------|---------------------------------------------------------------|
+| `EXEC_HINT_REQUEST_TYPE` | `0x03` | The [EIP-7685](./eip-7685.md) type prefix for execution hints |
+
+The [EIP-7685](./eip-7685.md) encoding of an execution hint request is as follows. 
+
+```python
+request_type = EXEC_HINT_REQUEST_TYPE
+request_data = rlp(txIndex, ...)
+```
+
+Where `txIndex` is an integer, 0 based index into the block transaction list.
+
+Clients MAY interpret the list of `txIndex` as the set of transactions in the block which may be executed in any order.
+
+Clients MUST execute the remainder of the transactions in the block, which do not appear in the request, in the order they appear in the block.
+
+Clients MUST NOT include this request if none of the transactions in the block are independent.
+
+In the case of an incorrect stateroot after parallel execution, clients MUST re-execute all transactions in the block in block order, to confirm the bad block before reporting it as such.
+
+## Rationale
+
+
+
+## Backwards Compatibility
+
+The use of general purpose requests in this design ensures maximal backward compatibility. The optional nature of the execution hint means clients may choose to ignore the entire request.
+
+## Test Cases
+
+TBD
+
+## Reference Implementation
+
+TBD
+
+## Security Considerations
+
+Proposers could include false execution maps, resulting in wasted work among clients who choose to use them.
```

Comment:
There is also no limit to the request so it could be filled to arbitrary bytes or have the same tx in it multiple times :thinking: I think an optional "helper network" would thus work here, and if nodes misbehave there then kick them out

Reply:
What incentive (aside from altruism) is there to provide these maps?  What incentive is there to provide correct and complete maps?

Said another way, If I don't send a map, or send an incomplete map what penalty would I suffer?  If I send an incorrect map what penalty would I suffer?

If these were added to the block header then perhaps the gas available for a block could be extended by some amount of TXes that are also parallel?  (C * log(lanes) * avg gas-per-lane)?  And if there are any cross lane conflicts the whole TX is invalod?

Reply:
A strong security requirement for EL clients that support parallel execution would be detecting the cases when transactions set for parallel execution aren’t independent. If there is no such detection then the result of block validation becomes non-deterministic (e.g. root can be valid or invalid depending on the local execution of each node), moreover a portion of the network may accept an invalid state root by accidentally repeating the order of execution used by a payload builder